src/flp/szk.rs

```diff
@@ -320,9 +321,10 @@ where
 {
     /// Construct an instance of this sharedZK proof system with the underlying
     /// FLP.
-    pub fn new(typ: T) -> Self {
+    pub fn new(typ: T, algorithm_id: u32) -> Self {
         Self {
             typ,
+            id: algorithm_id.to_le_bytes(),
```

Comment:
Note that Prio3 and Poplar1 are using big-endian encoding for the algorithm ID in their binder strings. It might be nice to change the spec, so that the byte orders match.

Reply:
Good call: https://github.com/jimouris/draft-mouris-cfrg-mastic/pull/112

I'll add this to the top of the stack.